### ISSUE TITLE
Skip rejected vulnerabilities even when Red Hat doesn't say so

### ISSUE THREAD

**Issue by knqyf263** [aquasecurity/trivy-db#60]
In the case of CVE-2018-10754, Red Hat doesn't say it is rejected.
https://access.redhat.com/security/cve/CVE-2018-10754

But NVD describes it is rejected.
https://nvd.nist.gov/vuln/detail/CVE-2018-10754

In this case, we have to skip this vulnerability even if Red Hat doesn't say so.

**Comment by knqyf263** [comment]
@rahul2393 Could you look into it?

This PR just skips inserting a vulnerability from NVD, but Red Hat still inserts the vulnerability.
https://github.com/aquasecurity/trivy-db/pull/57

Here are the steps in trivy-db:
- Insert vulnerabilities of NVD into trivy-db
  - https://github.com/aquasecurity/trivy-db/tree/master/pkg/vulnsrc/nvd
- Insert vulnerabilities of Red Hat into trivy-db
  - https://github.com/aquasecurity/trivy-db/tree/master/pkg/vulnsrc/redhat
- Insert vulnerabilities...
- They have duplicate data like title, description, and references. To remove them, trivy-db optimizes all buckets.
  - https://github.com/aquasecurity/trivy-db/blob/master/pkg/vulnsrc/vulnsrc.go#L147

I think it is difficult to skip inserting rejected vulnerabilities in the Red Hat step because the process doesn't know NVD data. We probably have to skip rejected vulnerabilities in the optimization. Note that we have two optimizers.
https://github.com/aquasecurity/trivy-db/blob/master/pkg/vulnsrc/vulnsrc.go#L183
https://github.com/aquasecurity/trivy-db/blob/master/pkg/vulnsrc/vulnsrc.go#L209

If you have any other idea, let me know, please.

**Comment by rahul2393** [comment]
Sure.

**Comment by rahul2393** [comment]
@knqyf263 I agree with you, I think instead of rejecting vulnerability at the source level (which we are doing currently), at optimizer level we need to mark a vulnerability rejected.
